docs/src/main/tut/typeclasses/abstract_algebra.md

```diff
@@ -1,75 +1,29 @@
-# Abstract Algebra
-
-Before getting into the details of Algebird, let's go through a few simple abstract algebra definitions.  Folks interested in more details may wish to refer to the excellent [*Notes on Abstract Algebra*](http://www.math.usm.edu/perry/old_classes/mat423fa11/notes_25aug2011.pdf) from University of Southern Mississipi professor John Perry.
-
-## Semigroup
-
-Given a set S and an operation `+`, we say that `(S, +)` is a *semigroup* if it satisfies the following properties for any x, y, z &isin; S:
-
-- *Closure*: `x + y` &isin; S
-- *Associativity*: `(x + y) + z = x + (y + z)`
-
-We also say that *S forms a semigroup under **.
-
-#### Examples of Semigroups
-
-- Strings form a semigroup under concatenation.
-
-## Monoids
-
-A monoid is a semigroup with an identity element.  More formally, given a set M and an operation `+`, we say that `(M, +)` is a *monoid* if it satisfies the following properties for any x, y, z &isin; M:
-
-- *Closure*: `x + y` &isin; M
-- *Associativity*: `(x + y) + z = x + (y + z)`
-- *Identity*: There exists an e &isin; M such that `e + x = x + e = x`
+---
+layout: docs
+title:  "Abstract Algebra"
+section: "typeclasses"
```

Comment:
and now this is going to become a discussion of why abstract algebra is useful in programming